int32 reference_object_id
---
int32 target_object_id
